src/http/httpd.c

```diff
@@ -125,23 +151,52 @@ void http_send_response_chunk(struct http_client *hc, const void *chunk, size_t
     if (!hc) return;
     snprintf(txt_chunk, sizeof(txt_chunk), "%zx\r\n", len);
     if (hc->ssl) {
-        wolfSSL_write(hc->ssl, txt_chunk, strlen(txt_chunk));
-        wolfSSL_write(hc->ssl, chunk, len);
-        wolfSSL_write(hc->ssl, "\r\n", 2);
+        int rc = wolfSSL_write(hc->ssl, txt_chunk, strlen(txt_chunk));
+        if (rc <= 0)
+            goto close_conn;
+        rc = wolfSSL_write(hc->ssl, chunk, len);
+        if (rc <= 0)
+            goto close_conn;
+        rc = wolfSSL_write(hc->ssl, "\r\n", 2);
+        if (rc <= 0)
+            goto close_conn;
     } else {
         struct wolfIP *s = hc->httpd->ipstack;
-        wolfIP_sock_send(s, hc->client_sd, txt_chunk, strlen(txt_chunk), 0);
-        wolfIP_sock_send(s, hc->client_sd, chunk, len, 0);
-        wolfIP_sock_send(s, hc->client_sd, "\r\n", 2, 0);
+        int rc = wolfIP_sock_send(s, hc->client_sd, txt_chunk, strlen(txt_chunk), 0);
+        if (rc <= 0)
+            goto close_conn;
+        rc = wolfIP_sock_send(s, hc->client_sd, chunk, len, 0);
+        if (rc <= 0)
+           goto close_conn;
+        rc = wolfIP_sock_send(s, hc->client_sd, "\r\n", 2, 0);
+        if (rc <= 0)
+            goto close_conn;
+    }
```

Comment:
I think these sections could be simplified a lot if you had a function like `wolfIP_write` that would internally detect if ssl or plaintext should be used. Maybe this function could also handle `WANT_WRITE`.